File: benchmark/web/javascript/nest-10-express-nodejs-16/src/app.controller.ts
import {Controller, Get, Query} from '@nestjs/common';
import * as http from "http";

// fetch() is not available prior Node.js 18, and it is slower than http.get()
// TODO: Use fetch() as soon as the following issue is resolved:
// https://github.com/nodejs/undici/issues/1203
function fetch(url: string) {
  return new Promise((resolve, reject) => {
    http.get(url, (res) => {
      let data = '';

      res.on('data', (chunk) => {
        data += chunk;
      });

      res.on('end', () => {
        resolve(JSON.parse(data));
      });
    }).on('error', (err) => {
      reject(err);
    });
  });
}

@Controller('api/v1/periodic-table')
export class AppController {
  @Get('element')
  async getElement(@Query('symbol') symbol: string) {
    const elementRes = await fetch('http://web-data-source/element.json');
    const element = elementRes[symbol];

    return {
      name: element.name,
      number: element.number,
      group: element.group
    };
  }

  @Get('shells')
  async getShells(@Query('symbol') symbol: string) {
    const shellsRes = await fetch('http://web-data-source/shells.json');

    return {
      shells: shellsRes[symbol]
    };
  }
}

File: benchmark/web/dart/serinus-0.3.1-aot-3.3/lib/app_controller.dart
import 'package:serinus/serinus.dart';

import 'app_routes.dart';
import 'app_provider.dart';

class AppController extends Controller {

  AppController({super.path = '/'}){
    on(BenchRoute(path: '/api/v1/periodic-table/element'), _handleElement);
    on(BenchRoute(path: '/api/v1/periodic-table/shells'), _handleShells);
  }

  Future<Response> _handleElement(RequestContext context) async {
    final String symbol = context.queryParameters['symbol'];
    final Map<String, dynamic> element = await context.use<AppProvider>().getElement(symbol);

    return Response.json(element);
  }

  Future<Response> _handleShells(RequestContext context) async {
    final String symbol = context.queryParameters['symbol'];
    final Map<String, dynamic> shells = await context.use<AppProvider>().getShells(symbol);
    return Response.json(shells);
  }

}